poetry/factory.py

```diff
@@ -72,7 +72,22 @@ def create_poetry(
         # Configuring sources
         sources = poetry.local_config.get("source", [])
         for source in sources:
-            repository = self.create_legacy_repository(source, config)
+            # If a source has an mirror, use it.
+            if source.name in existing_repositories:
```

Comment:
This is another silly error I made. `source` is a dict and should be accessed using source["name"]. Maybe I programmed in JavaScript too much.